boofuzz/monitors/network_monitor.py

```diff
@@ -0,0 +1,55 @@
+from .imonitor import IMonitor
+from . import pedrpc
+
+# Important: IMonitor needs to come *before* pedrpc.Client in the
+# Inheritance list for the method resolution order to produce
+# correct results.
+
+
+class NetworkMonitor(IMonitor, pedrpc.Client):
+    """
+    Proxy class for the network monitor interface.
+
+    In Versions < 0.2.0, boofuzz had network and process monitors
+    that communicated over RPC. The RPC client was directly passed
+    to the session class, and resolved all method calls dynamically
+    on the RPC partner.
+
+    Since 0.2.0, every monitor class must implement the abstract class
+    IMonitor, which defines a common interface among all Monitors. To
+    aid future typehinting efforts and to make Network- and Process Monitors
+    disambiguable, this explicit proxy class has been introduced that
+    fast-forwards all calls to the RPC partner.
+
+    .. versionadded:: 0.2.0
+    """
+
+    def __init__(self, host, port):
+        super(IMonitor, self).__init__()
+        super(pedrpc.Client, self).__init__(host, port)
+
+    def pre_send(self, target=None, fuzz_data_logger=None, session=None):
+        return super(pedrpc.Client, self).pre_send(session.total_mutant_index)  # pytype: disable=attribute-error
```

Comment:
Ignoring attribute errors here makes it hard to tell, but I'm still not fully convinced that this will work.
Couldn't test it as there are previous errors. If you have a working test case please share it.

Reply:
It should now. The call is explicitely delegated to pedrpc.Client, which triggers a call to __getattr__ which forwards the call to a __method_missing method which pickles everything and sends it to the RPC server. I did not touch this part, so as long as it worked before (which I'll assume) it should work now.